src/QMCWaveFunctions/Jastrow/BsplineFunctor.h

```diff
@@ -658,7 +658,7 @@ BsplineFunctor<T>::evaluateV(const int iat, const int iStart, const int iEnd,
 #pragma vector always 
   for ( int jat = 0; jat < iLimit; jat++ ) {
     real_type r = distArray[jat];
-    if ( r < cutoff_radius )
+    if ( r < cutoff_radius && iStart+jat != iat )
```

Comment:
What is the additional logic for here?  Probably worth an in code comment.

Reply:
This a very important change to facilitate virtual moves. Previously, the exclusion of self iteration term is done at the distance table level. During a pbyp move, the iel-iel distance was assigned a very big distance. However, in a virtual move, you will compute the ratio with respect to every electron. So the distance table must contain intact distances (see the changes in distance table SoA classes). So the removal of self interation term must be handled at evaluation.